src/resources/templates/baseline/proxy/Caddyfile.j2

```diff
@@ -2,71 +2,24 @@
 	email admin@brightsparklabs.com
 
 	log {
-		level {{ settings.team_bsl.debug.proxy.log_level|lower }}
+		level {{ settings.reverse_proxy.debug.proxy.log_level|lower }}
 	}
 
-	{% if settings.team_bsl.services.proxy.enable_staging_ca %}
+	{% if settings.reverse_proxy.services.proxy.enable_staging_ca %}
 	acme_ca https://acme-staging-v02.api.letsencrypt.org/directory
-	{% endif %}{# settings.team_bsl.services.proxy.enable_staging_ca #}
+	{% endif %}{# settings.reverse_proxy.services.proxy.enable_staging_ca #}
 
-	{% if settings.team_bsl.services.proxy.enable_internal_tls %}
+	{% if settings.reverse_proxy.services.proxy.enable_internal_tls %}
 	local_certs
-	{% endif %}{# settings.team_bsl.services.proxy.enable_internal_tls #}
+	{% endif %}{# settings.reverse_proxy.services.proxy.enable_internal_tls #}
 
-	{% if settings.team_bsl.services.proxy.acme_dns_challenge.enable %}
-	acme_dns route53 {
-		access_key_id "{{ settings.team_bsl.services.proxy.acme_dns_challenge.access_key_id }}"
-		secret_access_key "{{ settings.team_bsl.services.proxy.acme_dns_challenge.secret_access_key }}"
-	}
-	{% endif %}{# settings.team_bsl.services.proxy.enable_acme_dns_challenge #}
 }
 
-https://bslnet.{{ settings.team_bsl.services.proxy.domain }} {
-	# Default to landing page.
-	redir / /landing-page/
-	reverse_proxy /landing-page/* http://landing-page:8080
-
-	redir /plummet /plummet/index.html
-	reverse_proxy /plummet/* http://plummet
-
-	redir /whoami /whoami/
-	reverse_proxy /whoami/* http://whoami {
-		header_up Host {host}
-		header_up X-Real-IP {remote_host}
-	}
-}
 
-https://docker.{{ settings.team_bsl.services.proxy.domain }} {
-	reverse_proxy http://docker-registry:5000
-}
+{% for route in settings.reverse_proxy.routing.routes %}
 
-https://docker-registry-ui.bslnet.{{ settings.team_bsl.services.proxy.domain }} {
-	reverse_proxy http://docker-registry-ui
+https://{{ route.dns_record }} {
```

Comment:
I want the whole thing to be templated, so you COULD specify http if you really want:

```
{{ route.external_endpoint }} {
  reverse_proxy {{ route.internal_endpoint}}
}
```